python/dlr/counter/model_exec_counter.py

```diff
@@ -0,0 +1,39 @@
+import threading
+import copy
+
+lock = threading.Lock()
+
+class ModelExecCounter(object):
+    model_dict = {}
+    intr_dict = {}
+    INIT = 1
+    INCREMENT = 2
+    CLEAR = 3
+    GETINTDICT = 4
+
+    @staticmethod
+    def update_model_run_count(model):
+        """keep counting model inference in dictionary"""
+        cnt = ModelExecCounter.model_dict.get(model)
+        if cnt:
+            cnt += 1
+            ModelExecCounter.update_dict(ModelExecCounter.INCREMENT, model, cnt)
+        else:
+            ModelExecCounter.update_dict(ModelExecCounter.INIT, model)
+
+    @staticmethod
+    def update_dict(op, model=None, cnt=1):
+        with lock:
+            if op == ModelExecCounter.INIT:
+                ModelExecCounter.model_dict[str(model)] = 1
+            elif op == ModelExecCounter.INCREMENT:
+                ModelExecCounter.model_dict[str(model)] = cnt
+            elif op == ModelExecCounter.GETINTDICT:
+                ModelExecCounter.intr_dict = copy.deepcopy(ModelExecCounter.model_dict)
```

Comment:
I assume this is for dictionary size change exception, correct?

Instead of introducing a new enum, can we just copy a list of keys in model_metrics.py? https://stackoverflow.com/questions/11941817/how-to-avoid-runtimeerror-dictionary-changed-size-during-iteration-error

Reply:
I have considered below use case, 1. To unblock the main thread to continuous update primary dictionary, we created a temporary dictionary (using deep copy instead of shallow copy).
- This way the main thread will not wait until Publisher read items from the primary dictionary.
- Easy to share data between CounterMgr and Publisher